homeassistant/components/influxdb/__init__.py

```diff
@@ -55,98 +65,155 @@
 BATCH_TIMEOUT = 1
 BATCH_BUFFER_SIZE = 100
 
-COMPONENT_CONFIG_SCHEMA_ENTRY = vol.Schema(
-    {vol.Optional(CONF_OVERRIDE_MEASUREMENT): cv.string}
+DB_CONNECTION_FAILURE_MSG = ()
+
+COMPONENT_CONFIG_SCHEMA_CONNECTION = {
+    # Connection config for V1 and V2 APIs.
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Optional(CONF_PATH): cv.string,
+    vol.Optional(CONF_PORT): cv.port,
+    vol.Optional(CONF_SSL): cv.boolean,
+    # Connection config for V1 API only.
+    vol.Inclusive(CONF_USERNAME, "authentication"): cv.string,
+    vol.Inclusive(CONF_PASSWORD, "authentication"): cv.string,
+    vol.Optional(CONF_DB_NAME, default=DEFAULT_DATABASE): cv.string,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    # Connection config for V2 API only.
+    vol.Inclusive(CONF_API_V2, "v2_api_config"): True,
+    vol.Inclusive(CONF_TOKEN, "v2_api_config"): cv.string,
+    vol.Inclusive(CONF_ORG, "v2_api_config"): cv.string,
+    vol.Optional(CONF_BUCKET, default=DEFAULT_BUCKET): cv.string,
+}
+
+_CONFIG_SCHEMA_ENTRY = vol.Schema({vol.Optional(CONF_OVERRIDE_MEASUREMENT): cv.string})
+
+_CONFIG_SCHEMA_PLATFORM = vol.Schema(
```

Comment:
What does platform mean here?

Reply:
Shouldn't be platform, you're right. I did a bunch of refactoring so validation logic for config was shared when possible between the sensor and the recorder components. I think I borrowed the platform term from sensor but it doesn't really make sense here.  I'll change it to `_CONFIG_SCHEMA` and drop `PLATFORM`.